client/faucet/src/main.rs

```diff
@@ -313,6 +333,27 @@ mod tests {
         );
     }
 
+    fn get_trade_ids_from_payload(
+        payload: &diem_types::transaction::TransactionPayload,
+    ) -> Vec<String> {
+        match payload {
+            Script(script) => match ScriptCall::decode(script) {
+                Some(ScriptCall::PeerToPeerWithMetadata { metadata, .. }) => {
+                    match bcs::from_bytes(&metadata).expect("should decode metadata") {
+                        Metadata::CoinTradeMetadata(CoinTradeMetadata::CoinTradeMetadataV0(
+                            coin_trade_metadata,
+                        )) => {
+                            return coin_trade_metadata.trade_ids;
+                        }
+                        _ => panic!("unexpected type of transaction metadata"),
+                    }
+                }
+                _ => panic!("unexpected type of script"),
+            },
+            _ => panic!("unexpected payload type"),
+        }
+    }
+
```

Comment:
This function would be much easier to read if it returned a `Result`, that way you could use ? to make it a bit more readable, and then if you wanted to panic the caller can make that decision (with an expect or unwrap call)

Reply:
So I didn't realize this was test code, so this is probably ok since for test code we want to panic early anyway.